app/entitycore_helper.py

```diff
@@ -0,0 +1,334 @@
+        val_details_out_dir (str or Pathlib.Path or None): directory to save
+            the validation details files.
+    """
+    if val_details_out_dir is None:
+        val_details_out_dir = pathlib.Path("./validation_details") / memodel.name
+    else:
+        val_details_out_dir = pathlib.Path(val_details_out_dir)
+    val_details_out_dir.mkdir(parents=True, exist_ok=True)
+    for key, val_dict in validation_dict.items():
+        # not a validation
+        if key == "memodel_properties":
+            continue
+        # do not register ValidationResult if it already exists
+        # Once we are able to delete the ValidationResult, we should move to the following logic:
+        # delete the ValidationResult if it already exists
+        # register the new one
+        iterator = client.search_entity(
+            entity_type=ValidationResult,
+            query={"name": val_dict["name"], "validated_entity_id": memodel.id},
+            token=access_token,
+        )
+        cal = iterator.first()
+        if cal is not None:
+            continue
+
+        # register validation result
+        validation_result = ValidationResult(
+            name=val_dict["name"],
+            passed=val_dict["passed"],
+            validated_entity_id=memodel.id,
+        )
+        registered = client.register_entity(
+            entity=validation_result,
+            token=access_token,
+        )
+
+        # register figure(s) as asset(s)
+        for fig_path in val_dict["figures"]:
+            client.upload_file(
+                entity_id=registered.id,
+                entity_type=ValidationResult,
+                file_path=str(fig_path),
+                file_content_type=f"application/{str(fig_path).split('.')[-1]}",
+                token=access_token,
+            )
+
+        if val_dict["validation_details"]:
+            # write down validation details to a file
+            val_details_fname = f"{val_dict['name'].replace(' ', '')}_validation_details.txt"
+            val_details_path = val_details_out_dir / val_details_fname
+            with open(val_details_path, "w") as f:
+                f.write(val_dict["validation_details"])
+            # register validation details as asset
+            client.upload_file(
+                entity_id=registered.id,
+                entity_type=ValidationResult,
+                file_path=str(val_details_path),
+                file_content_type="application/txt",
+                token=access_token,
+            )
+
+
+def run_and_save_calibration_validation(client, access_token, memodel_id=None, memodel_name=None):
+    """Download MEModel, run MEModel validation, and save validation and calibration results.
+
+    Args:
+        client (Client): EntitySDK client
+        access_token (str): access token for authentication
+        memodel_id (str or None): id of the MEModel to download
+        memodel_name (str or None): name of the MEModel to download
```

Comment:
Are the names unique? Because if not - fetching by name might give unexpected results.

Reply:
Yeah, I guess it's best to, not to confuse whomever will be working with this in the future.
Or we add a comment + log warning that this is not meant for production use with some explanation.

Reply:
I removed it in latest commit